openmc/mesh.py

```diff
@@ -433,6 +433,9 @@ def lower_left(self, lower_left):
         cv.check_length('mesh lower_left', lower_left, 1, 3)
         self._lower_left = lower_left
 
+        if self.is_flat():
+            raise ValueError("mesh cannot be flat")
```

Comment:
```suggestion
        if self.upper_right is not None and any(np.isclose(upper_right, lower_left)):
            raise ValueError("mesh cannot have zero thickness is any dimension")
```

Reply:
You could do a private method, but I think a one line check in the two setters is cleaner

Reply:
Fine with me!

Reply:
The only issue with this is tolerance:

```python
    mesh = openmc.RegularMesh()
    mesh.lower_left = [-25, -25, -25]
    mesh.upper_right = [-25+1e-6, -25, -25]
```

Is considered as a zero-thickness mesh. Don't know if that's an issue or not.